src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -1160,6 +1168,13 @@ private unsafe SocketError FinishOperationConnect()
         {
             try
             {
+                if (_currentSocket!.SocketType != SocketType.Stream)
```

Comment:
@antonfirsov @scalablecory also give this condition some thought. Personally, I'm fine if CI passes. You may have a better understanding of what `SO_UPDATE_CONNECT_CONTEXT` does to decide if this is the most appropriate check.

Reply:
This should match the logic for `CanUseConnectEx`. We need to call `SO_UPDATE_CONNECT_CONTEXT` if `ConnectEx` is used.

Reply:
I was thinking the same thing, though it seems the stream unix socket, which doesn't support ConnectEx, doesn't give an error when we set this socket option.
We added it because the UDP socket gave an error.